src/main/scala/beam/utils/scenario/generic/writers/ParkingInfoWriter.scala

```diff
@@ -0,0 +1,80 @@
+package beam.utils.scenario.generic.writers
+
+import beam.utils.csv.CsvWriter
+import beam.utils.data.synthpop.GeoService
+import beam.utils.data.synthpop.models.Models.{GenericGeoId, TazGeoId}
+import com.vividsolutions.jts.geom.Geometry
+import org.geotools.geometry.jts.JTS
+import org.geotools.referencing.CRS
+import org.opengis.referencing.operation.MathTransform
+
+import scala.util.Try
+
+trait ParkingInfoWriter {
+  def write(path: String, geoService: GeoService, tazCounts: Map[GenericGeoId, (Int, Int)]): Unit
+}
+
+object CsvParkingInfoWriter extends ParkingInfoWriter {
+  private val headers: Array[String] =
+    Array("taz", "parkingType", "pricingModel", "chargingType", "numStalls", "feeInCents", "reservedFor")
+  val destinationCoordSystem = CRS.decode("EPSG:3395", true)
+  val sourceCoordSystem = CRS.decode("EPSG:4326", true)
+  val mt: MathTransform = CRS.findMathTransform(sourceCoordSystem, destinationCoordSystem, true)
+
+  override def write(path: String, geoService: GeoService, tazCounts: Map[GenericGeoId, (Int, Int)]): Unit = {
+    val csvWriter = new CsvWriter(path, headers)
+    try {
+      tazCounts.foreach {
+        case (tazGeoId, (numberPersons, numberWorkers)) =>
+          val scaledTazArea = geoService.tazGeoIdToGeom.get(tazGeoId.asInstanceOf[TazGeoId]) match {
```

Comment:
Now I see why you `GenericGeoId`. I left `PumaLevelScenarioGenerator.scala` just to have it as some reference that we did something on Puma level. If it starts to create so much headache, maybe we can stop having it as one of implementation of `ScenarioGenerator`

Reply:
Might be worth keeping in in case we move over to TAZs, but I haven't tried it on Puma IDs so there might be other ways that switching geometry breaks it